src/output/debug.rs

```diff
@@ -0,0 +1,156 @@
+// Copyright 2022 Rigetti Computing
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::{Error, OutputFormat};
+use crate::RecordedOutput;
+
+use qcs::ExecutionResult;
+
+/// Formats output of QIR programs in a debug-friendly structure.
+#[allow(clippy::module_name_repetitions)]
+#[derive(Debug, Default)]
+pub struct DebugOutputFormat(pub Vec<String>);
+
+impl OutputFormat for DebugOutputFormat {
+    /// Create an [`DebugOutputFormat`].
+    /// Will return [`enum@Error`] if something about the [`ExecutionResult`] or [`RecordedOutput`] is
+    /// unsupported, or if any of the result's data is indexed out-of-range.
+    ///
```

Comment:
```suggestion
    ///
    /// # Arguments
    ///
    /// - `result`: The output returned from the [`qcs`] crate.
    /// - `mapping`: The [`crate::ProgramOutput::recorded_output`] field from a call to [`crate::transpile_qir_to_quil`]
```

Maybe not useful in the context of this crate, but that was the sort of thing I was trying to do in my own docstrings (thus needing the re-export of `ProgramOutput`. I have not actually tested this docstring so my links could be off.

Reply:
just tested w/ cargo doc and it links up nicely -- added!